src/lib.rs

```diff
@@ -94,7 +95,9 @@ impl SessionStore for MongodbSessionStore {
         let filter = doc! { "session_id": id };
         match coll.find_one(filter, None).await? {
             None => Ok(None),
-            Some(doc) => Ok(Some(serde_json::from_str(doc.get_str("session")?)?)),
+            Some(doc) => Ok(Some(
+                bson::from_bson::<Session>(doc.get("session").unwrap().clone()).unwrap(),
```

Comment:
this probably should be ? instead of unwrap — if something goes wrong with the data state we wouldn't want to panic the server